common/persistence/persistenceHealthRateLimitedClients.go

```diff
@@ -0,0 +1,1384 @@
+		logger        log.Logger
+	}
+)
+
+var _ ShardManager = (*shardHealthRateLimitedPersistenceClient)(nil)
+var _ ExecutionManager = (*executionHealthRateLimitedPersistenceClient)(nil)
+var _ TaskManager = (*taskHealthRateLimitedPersistenceClient)(nil)
+var _ MetadataManager = (*metadataHealthRateLimitedPersistenceClient)(nil)
+var _ ClusterMetadataManager = (*clusterMetadataHealthRateLimitedPersistenceClient)(nil)
+var _ Queue = (*queueHealthRateLimitedPersistenceClient)(nil)
+
+func NewShardPersistenceHealthRateLimitedClient(persistence ShardManager, rateLimiter quotas.RequestRateLimiter, healthSignals backpressure.HealthSignalAggregator, logger log.Logger) ShardManager {
+	return &shardHealthRateLimitedPersistenceClient{
+		persistence:   persistence,
+		rateLimiter:   rateLimiter,
+		healthSignals: healthSignals,
+		logger:        logger,
+	}
+}
+
+func NewExecutionPersistenceHealthRateLimitedClient(persistence ExecutionManager, rateLimiter quotas.RequestRateLimiter, healthSignals backpressure.HealthSignalAggregator, logger log.Logger) ExecutionManager {
+	return &executionHealthRateLimitedPersistenceClient{
+		persistence:   persistence,
+		rateLimiter:   rateLimiter,
+		healthSignals: healthSignals,
+		logger:        logger,
+	}
+}
+
+func NewTaskPersistenceHealthRateLimitedClient(persistence TaskManager, rateLimiter quotas.RequestRateLimiter, healthSignals backpressure.HealthSignalAggregator, logger log.Logger) TaskManager {
+	return &taskHealthRateLimitedPersistenceClient{
+		persistence:   persistence,
+		rateLimiter:   rateLimiter,
+		healthSignals: healthSignals,
+		logger:        logger,
+	}
+}
+
+func NewMetadataPersistenceHealthRateLimitedClient(persistence MetadataManager, rateLimiter quotas.RequestRateLimiter, healthSignals backpressure.HealthSignalAggregator, logger log.Logger) MetadataManager {
+	return &metadataHealthRateLimitedPersistenceClient{
+		persistence:   persistence,
+		rateLimiter:   rateLimiter,
+		healthSignals: healthSignals,
+		logger:        logger,
+	}
+}
+
+func NewClusterMetadataPersistenceHealthRateLimitedClient(persistence ClusterMetadataManager, rateLimiter quotas.RequestRateLimiter, healthSignals backpressure.HealthSignalAggregator, logger log.Logger) ClusterMetadataManager {
+	return &clusterMetadataHealthRateLimitedPersistenceClient{
+		persistence:   persistence,
+		rateLimiter:   rateLimiter,
+		healthSignals: healthSignals,
+		logger:        logger,
+	}
+}
+
+func NewQueuePersistenceHealthRateLimitedClient(persistence Queue, rateLimiter quotas.RequestRateLimiter, healthSignals backpressure.HealthSignalAggregator, logger log.Logger) Queue {
+	return &queueHealthRateLimitedPersistenceClient{
+		persistence:   persistence,
+		rateLimiter:   rateLimiter,
+		healthSignals: healthSignals,
+		logger:        logger,
+	}
+}
+
+func (p *shardHealthRateLimitedPersistenceClient) GetName() string {
+	return p.persistence.GetName()
+}
+
+func (p *shardHealthRateLimitedPersistenceClient) GetOrCreateShard(
```

Comment:
were these generated by something? can they be generated by something? this seems hard to maintain if we ever want to change this interface